catroid/src/androidTest/java/org/catrobat/catroid/test/utils/CrashReporterTest.java

```diff
@@ -0,0 +1,161 @@
+		editor.commit();
+		CrashReporter crashReporter = new CrashReporter(context);
+		assertFalse(crashReporter.initialize());
+	}
+
+	@Test
+	public void testCrashlyticsInitializedOnAnonymousReportEnabled() {
+		Context context = InstrumentationRegistry.getTargetContext();
+		SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(context).edit();
+		editor.putBoolean(SettingsActivity.SETTINGS_CRASH_REPORTS, true);
+		editor.clear();
+		editor.commit();
+		CrashReporter crashReporter = new CrashReporter(context);
+		assertTrue(crashReporter.initialize());
+	}
+
+	@Test
+	public void testUnhandledExceptionStoredOnCrashReportEnabled() {
+
+		Exception exception = new RuntimeException("Error");
+		Context context = InstrumentationRegistry.getTargetContext();
+		CrashReporter crashReporter = new CrashReporter(context);
+		crashReporter.setIsCrashReportEnabled(true);
+		crashReporter.storeUnhandledException(exception);
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		SharedPreferences.Editor editor = sharedPreferences.edit();
+		editor.commit();
+		assertTrue(sharedPreferences.getString(CrashReporter.EXCEPTION_FOR_REPORT, "").length() > 0);
+	}
+
+	@Test
+	public void testUnhandledExceptionNotStoredOnCrashReportDisabled() {
+
+		Exception exception = new RuntimeException("Error");
+		Context context = InstrumentationRegistry.getTargetContext();
+		CrashReporter crashReporter = new CrashReporter(context);
+		crashReporter.setIsCrashReportEnabled(false);
+		crashReporter.storeUnhandledException(exception);
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		SharedPreferences.Editor editor = sharedPreferences.edit();
+		editor.commit();
+		assertTrue(sharedPreferences.getString(CrashReporter.EXCEPTION_FOR_REPORT, "").length() == 0);
+	}
+
+	@Test
+	public void testUnhandledExceptionStoredOnNoPreviousExceptionStored() {
+
+		Context context = InstrumentationRegistry.getTargetContext();
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		SharedPreferences.Editor editor = sharedPreferences.edit();
+
+		assertTrue(sharedPreferences.getString(CrashReporter.EXCEPTION_FOR_REPORT, "").length() == 0);
+
+		Exception exception = new RuntimeException("Error");
+		CrashReporter crashReporter = new CrashReporter(context);
+		crashReporter.setIsCrashReportEnabled(true);
+		crashReporter.storeUnhandledException(exception);
+		editor.commit();
+		assertTrue(sharedPreferences.getString(CrashReporter.EXCEPTION_FOR_REPORT, "").length() > 0);
+	}
+
+	@Test
+	public void testSharedPreferencesClearedAfterLoggingException() {
+
+		Context context = InstrumentationRegistry.getTargetContext();
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		SharedPreferences.Editor editor = sharedPreferences.edit();
+
+		Exception exception = new RuntimeException("Error");
```

Comment:
You can inline the variable (in such cases). IMO there is no need for a local variable.